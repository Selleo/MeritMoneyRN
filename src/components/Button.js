import React, { Component } from 'react'
import { Platform, TouchableOpacity, Text, StyleSheet, View } from 'react-native'
import LinearGradient from 'react-native-linear-gradient'
import { secondaryLight, primary, white, blueDark } from 'src/styles/colors'

export default class Button extends Component {
  static defaultProps = {
    onPress: () => {},
  }

  render() {
    const { text, outline, onPress } = this.props

    return (
      <TouchableOpacity onPress={onPress} style={styles.container}>
        <LinearGradient
          colors={[secondaryLight, primary]}
          end={{ x: 1, y: 1 }}
          locations={[0, 1]}
          start={{ x: 0, y: 1 }}
          style={styles.linearGradient}
        >
          <View style={[styles.textContainer, outline && styles.outline]} />
          <Text adjustsFontSizeToFit={true} style={styles.text}>
            {text}
          </Text>
        </LinearGradient>
      </TouchableOpacity>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    width: 150,
    height: 50,
  },
  linearGradient: {
    borderRadius: 25,
    zIndex: 1,
  },
  textContainer: {
    alignSelf: 'center',
    backgroundColor: 'transparent',
    borderRadius: 25,
    height: 42,
    position: 'absolute',
    top: 2.5,
    width: 142,
    zIndex: 2,
    ...Platform.select({
      ios: {
        height: 42,
      },
      android: {
        height: 38,
      },
    }),
  },
  outline: {
    backgroundColor: blueDark,
  },
  text: {
    textAlign: 'center',
    color: white,
    fontFamily: 'Lato-Regular',
    fontSize: 20,
    paddingVertical: 10,
    zIndex: 3,
  },
})
