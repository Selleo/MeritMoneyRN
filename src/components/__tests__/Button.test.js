import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import Button from '../Button'

describe('<Button />', () => {
  const defaultProps = {
    text: 'Click me',
    outline: false,
    onPress: jest.fn(),
  }

  const wrapper = renderer.create(<Button {...defaultProps} />)

  test('render', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
