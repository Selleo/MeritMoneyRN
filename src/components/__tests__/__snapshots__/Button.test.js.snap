// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> render 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 50,
      "opacity": 1,
      "width": 150,
    }
  }
>
  <BVLinearGradient
    colors={
      Array [
        4282147443,
        4281317757,
      ]
    }
    endPoint={
      Object {
        "x": 1,
        "y": 1,
      }
    }
    locations={
      Array [
        0,
        1,
      ]
    }
    startPoint={
      Object {
        "x": 0,
        "y": 1,
      }
    }
    style={
      Object {
        "borderRadius": 25,
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderRadius": 25,
            "height": 42,
            "position": "absolute",
            "top": 2.5,
            "width": 142,
            "zIndex": 2,
          },
          false,
        ]
      }
    />
    <Text
      accessible={true}
      adjustsFontSizeToFit={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "Lato-Regular",
          "fontSize": 20,
          "paddingVertical": 10,
          "textAlign": "center",
          "zIndex": 3,
        }
      }
    >
      Click me
    </Text>
  </BVLinearGradient>
</View>
`;
